### PR TITLE
This is a complete versioning system rework.

### LINKED ISSUE #554: Fix versioning file being committed with changes
Many people are committing the modified `version.txt`

Would be nice to have this not happen

### PR DESCRIPTION
**Pull request by dwaynepryce**
First, setuptools has pushed us toward using a `pyproject.toml` and `setup.cfg` in lieu of having a setuptools.setup() function call specify the package parameters.

In addition, we have a much easier way of injecting our prerelease tags. We still technically call `setuptools.setup()`, but it knows to rely on pyproject.toml and setup.cfg to provide all the parameters, but also to use a commandline argument for any possible version appendix to use.

However, this information is written to the sdist package, so the prerelease version numbers are maintained on publish. This is what we want.

If for some reason we are trying to get the version from source code, not the sdist package, we still try to use the `setup.cfg` file, and we append a dev tag based on today's date and time. This is most common for the developers who work on graspologic itself.

The thing this fixes is that using `pip install -r requirements.txt` no longer changes the source tree and include a then-materialized version number. This should avoid many of the accidental commits of `version.txt` that plagued and confused everyone in the past.

IMPORTANT NOTE: We may need to do some minor tweaks to fix the build workflow stuff because some of it isn't testable until we're on dev or main.

Closes #554

### CODE REVIEW

**Comment by dwaynepryce** [comment]
This also Closes #464

File: .github/workflows/build.yml
Comment (dwaynepryce) [inline .github/workflows/build.yml:100]: I may need to screw with this once we're on dev, we won't know if this works until we merge

File: graspologic/__init__.py
Comment (Nyecarr) [inline graspologic/__init__.py:22]: I am not sure why this was in the init originally but I agree that it is best practice to move it out to utils.py

File: setup.cfg
Comment (Nyecarr) [inline setup.cfg:10]: We should set up a maintainer distribution email similarly to how numpy does. Reduces single point of contact issues.

Comment (dwaynepryce) [reply setup.cfg:10]: I agree, I'm just not sure how we do that officially.

**Review by Nyecarr — APPROVED** [review]

**Pull request merged**
